plutus-core/cost-model/test/TestCostModels.hs

```diff
@@ -263,34 +190,150 @@ testPredictThree haskellModelFun modelR1 = propertyR $ do
   diff byR (~=) (predictH x y z)
 
 
--- TODO: discover the properties automatically.  Hedgehog's $$(discover) doesn't
--- work because it expects to find Properties, but our prop_* functions have
--- type RModels s -> Property.  I think this'll require us to steal some stuff
--- from Hedgehog.Internal.TH.
+testPredictSix
```

Comment:
I'm sure we should be able to make this stuff more generic... but not today

Reply:
> I'm sure we should be able to make this stuff more generic... but not today
Yes, that's what I thought too!  The makeProp ones are particularly annoying because the signatures are much bigger than the actual functions.  Unfortunately makeProp2 is different from the others because you have to tell it what region of the plane to do the testing on: that might go away at some point in the future though.